File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ctrMap = {}
        freq = [[] for i in range(len(nums) + 1)]
        for num in nums: 
            ctrMap[num] = ctrMap.get(num, 0 ) + 1
        for num, count in ctrMap.items():
            freq[count].append(num)
        
        res = []
        for i in range(len(nums), 0, -1):
            for num in freq[i]:
                res.append(num)
                if len(res) == k:
                    return res


        


        

File: Data Structures & Algorithms/duplicate-integer/submission-7.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        if( nums.length == 0 ){
            return false;
        }
        // Set<Integer> set = new HashSet<Integer>(); //on
        // for(int num : nums){
        //     if(set.contains(num)) return true; 
        //     else set.add(num);
        // }
        // return false; 
        return Arrays.stream(nums).distinct().count()< nums.length; 
    }
}